### DISCUSSION TITLE
Using the touch panel with interrupt

### DISCUSSION THREAD

**Discussion by mottibz** [Bodmer/TFT_eSPI#888] [Q&A - Touch] [answered]
Hello, I am using a 2.8" TFT via API that uses the ILI9341 (240 X 320) with an integrated touch panel. I am connecting it to an ESP32. The example code works fine but it works in polling mode which loads the SPI bus and the system for no reason. I am trying to do the same using interrupts. I connected the the T_IRQ pin to an ESP32 input and trying to only get touch data when I get an interrupt. The issue is that I constantly get interrupts even when I do not touch the screen.
The setup code that I use is:
  pinMode(TOUCH_IRQ, INPUT);
  touchIntSignal = false;
  attachInterrupt(digitalPinToInterrupt(TOUCH_IRQ), touchISR, RISING);
The loop code that I am using is:
  if (touchIntSignal) {
    touchIntSignal = false;

    Serial.println("touch interrupt occured");
  }
And I constantly get the message as if interrupts are occuring all the time. 
I tried setting the pinMode to INPUT_PULLUP and it did not help. I tried setting the interrupt to occur either on RISING, FALLING, CHANGE and it did not help either.
Can anyone help me with this?
Thanks a lot, Motti

**Comment by Bodmer** [comment, marked as answer]
Issue #820 contains helpful information. In particular note the [comment here](https://github.com/PaulStoffregen/XPT2046_Touchscreen#using-the-interrupt-pin--custom-use-would-preclude-the-normal-built-in-usage-the-warning-below-is-justified) about recurring interrupts and the solution.

**Reply by mottibz** [reply]
Thanks a lot for the info. That is what I found after digging into more materials. Need to ignore interrupts coming in while reading the touch data. All is working now. Thanks again!
